src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
     limitations under the License.
 ////
 
-= Log4j Web Applications
+= Using Log4j in a Jakarta EE environment
+:jetty-latest-url: https://jetty.org/docs/jetty/12
+:tomcat-latest-url: https://tomcat.apache.org/tomcat-11.0-doc
+:wildfly-latest-url: https://docs.wildfly.org/32
 
-== Using Log4j in Web Applications
+In a Jakarta EE environment, there are two possible approaches to logging:
 
-You must take particular care when using Log4j or any other logging framework within a Java EE web application. It's important for logging resources to be properly cleaned up (database connections closed, files closed, etc.) when the container shuts down or the web application is undeployed. Because of the nature of class loaders within web applications, Log4j resources cannot be cleaned up through normal means. Log4j must be "started" when the web application deploys and "shut down" when the web application undeploys. How this works varies depending on whether your application is a <<Servlet-3-0,Servlet 3.0 or newer>> or <<Servlet-2-5,Servlet 2.5>> web application.
+. Each application can use their own copy of Log4j Core and include `log4j-core` in the WAR or EAR archive.
+. Applications can also use a single copy of Log4j Core that must be installed globally on the application server.
 
-Due to the namespace change from `javax` to `jakarta`, you need to use `log4j-jakarta-web` instead of `log4j-web` for Servlet 5.0 or newer.
+While the first approach is the easiest to implement, it has some limitations:
 
-In either case, you'll need to add the `log4j-web` module to your deployment.
+* Log events emitted by each application and the libraries bundled with it will be handled by Log4j Core,
+but events related to the application emitted by a **shared** library (e.g. JPA implementation) will be handled by the application server.
+To diagnose a problem with the application you might need to look into multiple log files.
+* Each application must use a **different** log file to prevent problems with concurrent access to the same file by multiple applications.
+Problems may arise especially if xref:manual/appenders.adoc#rollingfileappender[rolling file appender] is used.
+* Web applications have a different lifecycle from the application server.
+Additional care is required to stop Log4j Core when the application is stopped.
+See <<log4j-jakarta-web>> for more details.
 
-[.note]
-====
-To avoid problems, the Log4j shutdown hook will automatically be disabled when the log4j-web jar is included.
+The second approach is more complex and requires changes to the configuration of the application server.
+See <<sharing>> for more details.
```

Comment:
The way 2nd approach is introduced strongly implies a preference towards the 1st one. On the contrary, I personally find the 1st approach a bad practice. Judging from my experience, logging should be ideally configured centrally in an application server. As a matter of fact, that is what I have seen most in companies. Am I mistaken with this observation?